tests/nixos/fetch-git/testsupport/setup.nix

```diff
@@ -0,0 +1,133 @@
+{ lib, config, extendModules, ... }:
+let
+  inherit (lib)
+    mkOption
+    types
+    ;
+
+  indent = lib.replaceStrings ["\n"] ["\n    "];
+
+  execTestCase = testCase: ''
+
+    ### TEST ${testCase.name}: ${testCase.description} ###
+
+    with subtest("${testCase.description}"):
+        repo = Repo("${testCase.name}")
+        ${indent testCase.script}
+  '';
+
+  # Used to export individual derivations for each test case
+  # this allows to run only a single test via:
+  #   `nix build .#hydraJobs.fetch-git.{test-case-name}`
+  makeSeparateTest = testCase: let
+    evaled = extendModules {
+      modules = [{
+        testCases = lib.mkForce [testCase];
+      }];
+    };
+  in
+    evaled.config.result;
```

Comment:
We should have such functionality upstream.

Reply:
Is that a requirement forgetting this PR merged?

Reply:
Not really. This is going to sound harsh, but you could remove it instead.
By doing this here, it feels like we're done, but we've really just created an ad hoc implementation that we wouldn't really want to develop new feature for. It deserves a better place than this! I'd be happy to review the same functionality in Nixpkgs. Meanwhile, you can unblock this PR by simplifying it.

Reply:
Alright, I removed the logic that exports the test cases as individual derivations